src/main/java/frc/robot/subsystems/internals/InternalSubsystem.java

```diff
@@ -145,20 +155,28 @@ public void periodic() {
             if (shotRequested /* && turretSubsystem.getState() == TurretSubsystem.ModuleState.HIGH_TOLERANCE
                  || rejecting && turretSubsystem.getState() == TurretSubsystem.ModuleState.LOW_TOLERANCE */) {
                 // Spin the top motor on a timer
-                exitTimer.start();
-                motorTop.set(0.5);
-
-                // If 1.5 seconds have elapsed, mark the shot as finished
-                if (exitTimer.hasElapsed(1.5)) {
-                    exitTimer.stop();
-                    exitTimer.reset();
-                    motorTop.set(0);
-
-                    // Reset states
+                if (!stagingDetected) {
```

Comment:
We shouldn't set `shotRequested` to false if there's not a ball in staging. The idea of `shotRequested` is that the driver tells internals to shoot by setting the boolean to true, and internals will fire the ball as soon as everything is ready; `shotRequested` is just a way of saying "please shoot the next ball you get". Setting it to false if there's not a staging ball would break being able to press it when internals isn't fully ready yet.